### PR TITLE
use digestif hashes

### PR DESCRIPTION
**Pull request by samoht**
Use Digestif.S instead of custom hash signature

The patch is too invasive to my taste, but I had to flatten a lot of the modules hiearchies to make everyting compile...

### CODE REVIEW

File: src/git/pack.ml [hunk 1]
```diff
@@ -1546,20 +1537,17 @@ struct
...
-module Stream
-    (Hash: S.HASH)
-    (Deflate: S.DEFLATE)
-= struct
-  module Entry = Entry(Hash)
-  module Delta = Delta(Hash)(Entry)
-  module Hunk = Hunk(Hash)
-  module Pack = Pack(Hash)(Deflate)(Entry)(Delta)(Hunk)
+module Stream (Hash: S.HASH) (Deflate: S.DEFLATE) = struct
+  module XEntry = Entry(Hash)
```
Comment (dinosaure) [inline src/git/pack.ml]: Why `X...` ?

File: src/git/s.ml [hunk 2]
```diff
@@ -106,33 +106,17 @@ module type DEFLATE = sig
...
+  include Digestif.S
 
-  val to_string : t -> string
-  val of_string : string -> t
-
-  type hex = string
+  val feed_cstruct: ctx -> Cstruct.t -> ctx
+  val compare: t -> t -> int
+  val hash: t -> int
+  val equal: t -> t -> bool
```
Comment (dinosaure) [inline src/git/s.ml]: We need to choose between `eq` and `equal` but have both is not a good idea. I think `equal` should be the best and I can change API of `Digestif.S` to provide `equal` instead `eq` but we need to choose one - then if we choose `equal`, we need to update this PR to replace any `H.eq` to `H.equal`.

**Comment by samoht** [comment]
Odd constraints on windows:

```

The following dependencies couldn't be met:
  - git-mirage -> git -> digestif >= 0.7.0 -> eqaf -> core_bench -> comparelib >= 109.27.00 -> type_conv < 112.02.00
  - git-mirage -> git -> digestif >= 0.7.0 -> eqaf -> core_bench -> core < v0.10
  - git-mirage -> git -> digestif >= 0.7.0 -> eqaf -> core_bench -> fieldslib = 109.20.00
  - git-mirage -> git -> digestif >= 0.7.0 -> eqaf -> core_bench -> sexplib < 113.25.00
  - git-mirage -> git -> digestif >= 0.7.0 -> eqaf -> core_bench -> textutils >= 112.17.00 -> sexplib < 113.25.00
  - git-mirage -> git -> digestif >= 0.7.0 -> eqaf -> core_bench -> textutils >= 112.17.00 -> core < v0.10
```

**Comment by samoht** [comment]
The windows issues should be fixed by https://github.com/ocaml/opam-repository/pull/12575

**Pull request merged**